### ISSUE TITLE
Transaction exception -- java.util.ConcurrentModificationException

### ISSUE THREAD

**Issue by tadas-subonis** [pushtorefresh/storio#478]
Hi,

I am trying to use StorIO in this manner:

```java
                        storIOSQLite.internal().beginTransaction();
                        try {
                            storIOSQLite
                                    .put()
                                    .objects(localStreamEntries)
                                    .withPutResolver(streamEntryPutResolver)
                                    .useTransaction(true)
                                    .prepare()
                                    .executeAsBlocking();

                            storIOSQLite
                                    .put()
                                    .objects(otherListWithOtherType)
                                    .withPutResolver(localStreamLoadingPutResolver)
                                    .useTransaction(true)
                                    .prepare()
                                    .executeAsBlocking();

                            storIOSQLite.internal().setTransactionSuccessful();
                        } finally {
                            storIOSQLite.internal().endTransaction();
                        }
```

And I get this exception
```
java.util.ConcurrentModificationException
            at java.util.HashMap$HashIterator.nextEntry(HashMap.java:787)
            at java.util.HashMap$KeyIterator.next(HashMap.java:814)
            at com.pushtorefresh.storio.sqlite.impl.DefaultStorIOSQLite$InternalImpl.notifyAboutPendingChangesIfNotInTransaction(DefaultStorIOSQLite.java:391)
            at com.pushtorefresh.storio.sqlite.impl.DefaultStorIOSQLite$InternalImpl.endTransaction(DefaultStorIOSQLite.java:435)
```

**Comment by artem-zinnatullin** [comment]
I'll prepare fix ASAP, thanks for report!

**Comment by artem-zinnatullin** [comment]
BTW, you don't need to write `useTransaction(true)` explicitly  because it's `true` by default :)

**Comment by tadas-subonis** [comment]
Thanks :)

**Comment by artem-zinnatullin** [comment]
Fixed by #479, sorry it was my stupid mistake in the code :(

It was not even related to concurrency.

**Comment by artem-zinnatullin** [comment]
@tadas-subonis we've just released StorIO v1.1.1, it should be available in about 10-20 minutes, thanks for report, sorry for bug and have a nice day!

**Comment by tadas-subonis** [comment]
That was very quick! Thanks!

On Tue, Aug 4, 2015 at 6:15 PM, Artem Zinnatullin <notifications@github.com>
wrote:

> @tadas-subonis <https://github.com/tadas-subonis> we've just released
> StorIO v1.1.1, it should be available in about 10-20 minutes, thanks for
> report, sorry for bug and have a nice day!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pushtorefresh/storio/issues/478#issuecomment-127645944>
> .
>



-- 

Kind Regards,
Tadas Šubonis
